### PR TITLE
expansion of exception stack trace

### PR DESCRIPTION
**Pull request by pelikhan**
- [ ] reverse lookup table sha -> script version

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -331,6 +339,49 @@ export async function getScript(
     )
 }
 
+const SHA_PART = "sha"
+function binarySha(hex: string) {
+    const s = createHash("sha256")
```
Comment (mmoskal) [inline src/storage.ts:344]:
```ts
if(!hex) return undefined
```

File: src/storage.ts [hunk 2]
```diff
@@ -331,6 +339,49 @@ export async function getScript(
...
+    scriptId: string,
+    scriptVersion: number,
+    sha: string
+) {
+    // check sha format, roundtrip throug buffer parser
+    sha = Buffer.from(sha, "hex").toString("hex")
+    // TODO: avoid sha collisions
+    await scriptVersionShaTable.upsertEntity(
+        {
+            partitionKey: SHA_PART,
```
Comment (mmoskal) [inline src/storage.ts:376]: I think it should use the same partition (user) key as the script

**Review by mmoskal — APPROVED** [review]

File: src/storage.ts (hunk 2 above)
Comment (pelikhan) [reply src/storage.ts:376]: ... but I don't know the script is for the lookup

Comment (mmoskal) [reply src/storage.ts:376]: messages from device have partitionId, so around `// TODO PARSE MESSAGE` you have `m.partitionId`

**Comment by mmoskal** [comment]
fine for me to hack this?

**Comment by pelikhan** [comment]
Yes take over
________________________________

**Pull request merged**